src/virtualship/cli/_plan.py

```diff
@@ -0,0 +1,1174 @@
+import datetime
+import os
+import traceback
+from typing import ClassVar
+
+from textual import on
+from textual.app import App, ComposeResult
+from textual.containers import Container, Horizontal, VerticalScroll
+from textual.screen import Screen
+from textual.validation import Function
+from textual.widgets import (
+    Button,
+    Collapsible,
+    Input,
+    Label,
+    Rule,
+    Select,
+    Static,
+    Switch,
+)
+
+from virtualship.cli.validator_utils import (
+    get_field_type,
+    group_validators,
+    is_valid_depth,
+    is_valid_lat,
+    is_valid_lon,
+    is_valid_timestr,
+    type_to_textual,
+)
+from virtualship.errors import UnexpectedError, UserError
+from virtualship.models.location import Location
+from virtualship.models.schedule import Schedule, Waypoint
+from virtualship.models.ship_config import (
+    ADCPConfig,
+    ArgoFloatConfig,
+    CTD_BGCConfig,
+    CTDConfig,
+    DrifterConfig,
+    InstrumentType,
+    ShipConfig,
+    ShipUnderwaterSTConfig,
+    XBTConfig,
+)
+from virtualship.models.space_time_region import (
+    SpatialRange,
+    TimeRange,
+)
+
+UNEXPECTED_MSG_ONSAVE = (
+    "Please ensure that:\n"
+    "\n1) All typed entries are valid (all boxes in all sections must have green borders and no warnings).\n"
+    "\n2) Time selections exist for all waypoints.\n"
```

Comment:
Is there also a way to delete waypoints in the tool?

Reply:
(or add)